app/collections/collaborators.jsx

```diff
@@ -0,0 +1,375 @@
+                  <td>{ROLES_INFO[role].description}</td>
+                </tr>
+              )}
+            </tbody>
+          </table>
+
+          <p>
+            <button type="submit" className="major-button" onClick={this.handleSubmit}>Add user role</button>
+          </p>
+        </form>
+      </div>
+    );
+  }
+}
+
+RoleCreator.defaultProps = {
+  collections: null
+};
+
+RoleCreator.propTypes = {
+  collection: React.PropTypes.shape({
+    id: React.PropTypes.string
+  }),
+  onAdd: React.PropTypes.func
+};
+
+export class RoleRow extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      owner: {},
+      saving: null
+    };
+
+    this.removeRoles = this.removeRoles.bind(this);
+    this.toggleRole = this.toggleRole.bind(this);
+    this.updateRoles = this.updateRoles.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.roleSet.get('owner')
+      .then((owner) => {
+        this.setState({ owner });
+      });
+  }
+
+  removeRoles() {
+    this.updateRoles([], this.props.onRemove);
+  }
+
+  toggleRole(role) {
+    const currentRoles = this.props.roleSet.roles;
+    const removing = currentRoles.includes(role);
+
+    if (removing) {
+      this.confirmDelete(currentRoles, role);
+    } else {
+      currentRoles.push(role);
+      this.updateRoles(currentRoles);
+    }
+  }
+
+  confirmDelete(currentRoles, role) {
+    alert((resolve) => {
+      const handleDelete = () => {
+        const index = currentRoles.indexOf(role);
+        currentRoles.splice(index, 1);
+
+        if (currentRoles.length) {
+          this.updateRoles(currentRoles);
+        } else {
+          this.removeRoles();
+        }
+        resolve();
+      };
+
+      return (
+        <div className="confirm-delete-dialog content-container">
+          <p>Are you sure you want to delete this role?</p>
+          <div>
+            <button className="minor-button" autoFocus={true} onClick={resolve}>No</button>
+            {' '}
+            <button className="major-button" onClick={handleDelete}>Yes</button>
+          </div>
+        </div>
+      );
+    });
+  }
+
+  updateRoles(newRoles = [], callback = () => {}) {
+    let promise;
+    this.setState({ saving: true });
+
+    if (newRoles.length > 0) {
+      promise = this.props.roleSet.update({ roles: newRoles }).save();
+    } else {
+      promise = this.props.roleSet.delete();
+    }
+
+    promise.then(() => {
+      callback();
+      this.setState({ saving: false });
+    })
+    .catch((error) => {
+      this.setState({ error });
+    });
+  }
+
+  render() {
+    const { owner } = this.state;
+
+    return (
+      <p>
+        {owner && (<strong>{owner.login}</strong>)}
+
+        { owner.display_name !== this.props.collectionOwner && (
+          <button type="button" className="pill-button" onClick={this.removeRoles}>Remove</button>
```

Comment:
@srallen @wgranger yep, that sounds good. Maybe `You are the collection owner` should be more bold so it doesn't get lost up there.
just as long as you, the collection owner, is treated differently to the other collection members and there's no chance of somehow removing yourself, the collection owner, from the collection.